src/psyclone/tests/psyGen_test.py

```diff
@@ -2668,17 +2692,609 @@ def test_binaryoperation_can_be_printed():
+    del ast2.content[0].content[2]
+    with pytest.raises(GenerationError) as error:
+        schedule = processor.generate_schedule("dummy_code", ast2)
+    assert "Unexpected kernel AST. Could not find " \
+           "subroutine: dummy_code" in str(error.value)
+
+
+def test_fparser2astprocessor_generate_schedule_two_modules():
+    ''' Tests the fparser2AST generate_schedule method raises an exception
+    when more than one fparser2 module node is provided.
+    '''
+    ast1 = fpapi.parse(FAKE_KERNEL_METADATA*2, ignore_comments=True)
+    metadata = DynKernMetadata(ast1)
+    my_kern = DynKern()
+    my_kern.load_meta(metadata)
+    ast2 = my_kern.ast
+    processor = Fparser2ASTProcessor()
+
+    # Test kernel with two modules
+    with pytest.raises(GenerationError) as error:
+        schedule = processor.generate_schedule("dummy_code", ast2)
+    assert ("Unexpected AST when generating 'dummy_code' kernel schedule."
+            " Just one module definition per file supported.") \
+        in str(error.value)
+
+
+def test_fparser2astprocessor_generate_schedule_dummy_subroutine():
+    ''' Tests the fparser2AST generate_schedule method with a simple
+    subroutine.
+    '''
+    dummy_kernel_metadata = '''
+    module dummy_mod
+      type, extends(kernel_type) :: dummy_type
+         type(arg_type), meta_args(3) =                    &
+              (/ arg_type(gh_field, gh_write,     w3),     &
+                 arg_type(gh_field, gh_readwrite, wtheta), &
+                 arg_type(gh_field, gh_inc,       w1)      &
+               /)
+         integer :: iterates_over = cells
+       contains
+         procedure, nopass :: code => dummy_code
+      end type dummy_type
+    contains
+     subroutine dummy_code(f1, f2, f3)
+        real(wp), dimension(:,:), intent(in)  :: f1
+        real(wp), dimension(:,:), intent(out)  :: f2
+        real(wp), dimension(:,:) :: f3
+        f2 = f1 + 1
+      end subroutine dummy_code
+    end module dummy_mod
+    '''
+    ast1 = fpapi.parse(dummy_kernel_metadata, ignore_comments=True)
+    metadata = DynKernMetadata(ast1)
+    my_kern = DynKern()
+    my_kern.load_meta(metadata)
+    ast2 = my_kern.ast
+    processor = Fparser2ASTProcessor()
+
+    # Test properly formed kernel module
+    schedule = processor.generate_schedule("dummy_code", ast2)
+    assert isinstance(schedule, KernelSchedule)
+
+    # Test argument intent is inferred when not available in the declaration
+    assert schedule.symbol_table.lookup('f3').scope == 'global_argument'
+    assert schedule.symbol_table.lookup('f3').is_input is True
+    assert schedule.symbol_table.lookup('f3').is_output is True
+
+    # Test that a kernel subroutine without Execution_Part still creates a
+    # valid KernelSchedule
+    del ast2.content[0].content[2].content[1].content[2]
+    schedule = processor.generate_schedule("dummy_code", ast2)
+    assert isinstance(schedule, KernelSchedule)
+    assert len(schedule.children) == 0
```

Comment:
This should be `assert not schedule.children`

Reply:
Done